include/utlist.h

```diff
@@ -292,30 +292,44 @@ do {
   head = add;                                                                                  \
 } while (0)
 
-#define LL_CONCAT(head1,head2)                                                                 \
-do {                                                                                           \
-  LDECLTYPE(head1) _tmp;                                                                       \
-  if (head1) {                                                                                 \
-    _tmp = head1;                                                                              \
-    while (_tmp->next) { _tmp = _tmp->next; }                                                  \
-    _tmp->next=(head2);                                                                        \
-  } else {                                                                                     \
-    (head1)=(head2);                                                                           \
-  }                                                                                            \
-} while (0)
+#define LL_CONCAT(head1, head2)                                               \
```

Comment:
utlist.h is an external header only lib. It's best we don't touch it for easier [synchs from upstream](https://troydhanson.github.io/uthash/utlist.html). Also, in this case it would be better to have new versions of the LL_CONCAT and LL_APPEND when atomic updates are needed (LL_CONCAT_ATOMIC, LL_APPEND_ATOMIC?) since they incur some overheads.

Reply:
utlist.h was previously modified to add  LL_COMPUTE_LENGTH (deac30a) that isn't present upstream. May I add LL_CONCAT_ATOMIC, LL_APPEND_ATOMIC, LL_FOREACH_ATOMIC to utlist.h or shall I move all the additional macros to a new file?

Reply:
OK, I didn't notice that. Yes it would be more handy if we kept our local additions in separate file and utlist.h identical with upstream.